### ISSUE TITLE
cq problem

### ISSUE THREAD

**Issue by jmblen** [chrisreyn/Tap-Titans-2-Progress-Bot#54]
ive been tweaking the sleep sections of the cq part of the bot.

where is the part that it checks that the bot is already in the clan boss stage and should start attacking?
i think that is the  root of the problem
 example
1 click-- to go boss stage
2 sleep
3 check if at boss stage
4 start attacking

rather the bot do this
1 click-- to go boss stage
2 no sleep(and there is a slight lag that turns the screen black in milliseconds before the boss stage loads)
3 starts check boss stage (checks the color of the black screen) = not at boss stage
4 clicks the uuper right part to close clan quest
= loops (starts the cq over and over again)
= eats diamonds

**Comment by chrisreyn** [comment]
it's somewhere here but im a bit busy to check which one :(
```
getColor #color1 420 725
	//420 730
	
	if #color1 == -1785765 and #clanQuestCount == 0
		#recheckClanQuest = 0
		goto :clanQuestClose
	endif
	
	//boss progress color 
	getColor #color1 175 335
	
	#clanQuestLoopCount = 0
	
	if #color1 == 2055160
		//boss up
		touchDown 0 310 740
		sleep #btnDelay
		touchUp 0
		sleep #menuPopUpDelay
		goto :clanQuestCheckDias
	elseif #color1 == 0
		//boss down
		#recheckClanQuest = 0
	endif

	//close (and recheck)
goto :clanQuestClose
```

**Comment by jmblen** [comment]
sleep 3000 ( I EDITED THIS SLEEP PART WHICH CHECKS IF AT CLAN BOSS STAGE)
		getColor #color1 20 770
		
		goto :clanQuestHit
	endif
	
:clanQuestHit
	if #color1 == -28929 and #clanQuestLoopCount == 0
		//add quest
		#clanQuestCount = #clanQuestCount + 1
	endif

	#clanQuestLoopCount = #clanQuestLoopCount + 1
	
	if #color1 == -28929 and #clanQuestLoopCount < 1000
		//hit the boss
		touchDown 0 180 730
		touchUp 0
		sleep 28
		goto :clanQuestHit
	elseif #clanQuestLoopCount > 950
		goto :clanQuestClose
endif


THIS CHANGE WILL ONLY APPLY IF YOU ARE ATTACKING ONCE PER UP OF CLAN QUEST BOSS
I TRIED ATTACK TWICE PER UP BUT I THINK THE SLEEP I CHANGE IN THE UPPER PART DOES NOT CONCERN THE 2nd CQ ATTACK
@chrisreyn

**Comment by chrisreyn** [comment]
@jmblen i will check tomorrow maybe. no false-positives so far?

**Comment by jmblen** [comment]
yep no flase positive on my memu emulator.. ive been reading other problems here,, and i can assume that it may only work for me,, to others it may or may not work

**Comment by chrisreyn** [comment]
@jmblen how's the CQ? still getting correct readings? if yes, can you comment your clan quest code. i couldnt follow your comment so i might as well copy paste it into mine. please enclose it between 2 triple backquotes (```) so it retains formatting
